symfony/framework-bundle/3.3/src/Kernel.php

```diff
@@ -58,4 +59,78 @@ protected function configureRoutes(RouteCollectionBuilder $routes)
         $routes->import($confDir.'/{routes}/'.$this->environment.'/**/*'.self::CONFIG_EXTS, '/', 'glob');
         $routes->import($confDir.'/{routes}'.self::CONFIG_EXTS, '/', 'glob');
     }
+
+    public static function bootstrapCli(array &$argv)
+    {
+        // consume --env and --no-debug from the command line
+
+        // when using symfony/console v4.2 or higher, this should
+        // be replaced by a call to Application::bootstrapEnv()
+
+        for ($i = 0; $i < \count($argv) && '--' !== $v = $argv[$i]; ++$i) {
+            if ('--no-debug' === $v) {
+                putenv('APP_DEBUG='.$_SERVER['APP_DEBUG'] = $_ENV['APP_DEBUG'] = '0');
+                $argvUnset[$i] = true;
+                break;
+            }
+        }
+
+        for ($i = 0; $i < \count($argv) && '--' !== $v = $argv[$i]; ++$i) {
+            if (!$v || '-' !== $v[0] || !preg_match('/^-(?:-env(?:=|$)|e=?)(.*)$/D', $v, $v)) {
+                continue;
+            }
+            if (!empty($v[1]) || !empty($argv[1 + $i])) {
+                putenv('APP_ENV='.$_SERVER['APP_ENV'] = $_ENV['APP_ENV'] = empty($v[1]) ? $argv[1 + $i] : $v[1]);
+                $argvUnset[$i] = $argvUnset[$i + empty($v[1])] = true;
+            }
+            break;
+        }
+
+        if (!empty($argvUnset)) {
+            $argv = array_values(array_diff_key($argv, $argvUnset));
+        }
+    }
+
+    public static function bootstrapEnv($env = null)
+    {
+        if (null !== $env) {
+            putenv('APP_ENV='.$_SERVER['APP_ENV'] = $env);
+        }
+
+        if ('prod' !== $_SERVER['APP_ENV'] = isset($_SERVER['APP_ENV']) ? $_SERVER['APP_ENV'] : (isset($_ENV['APP_ENV']) ? $_ENV['APP_ENV'] : null)) {
```

Comment:
undefined `APP_ENV` in dev mode generally means you forgot to create your local `.env` file from the dist one. Now it defaults to `dev` so the app may run while it misses project env vars, and you have to figure it out yourself. Not sure it's better DX wise?

Reply:
hmm forgot about it, it still throws